File: sieveOfEratosthenes.py
from math import sqrt

def primeNumbers(n):
    primes = [True] * (n + 1)

    primes[0] = False
    primes[1] = False

    sqrt_n = int(sqrt(n))

    for i in range(2, sqrt_n + 1):
        if (primes[i]):
            for j in range(i * 2, n + 1, i):
                primes[j] = False
                

    result = [i for i, b in enumerate(primes) if b]


    return result


print(primeNumbers(100000000))

File: sieveOfEratosthenes.js
const primeNumbers = n => {
  const primes = [];

  // initialize the array to true ie 1 in this case
  for (i = 0; i <= n; i++) {
    primes[i] = 1;
  }

  // 1 and 0 are not prime numbers
  primes[0] = 0;
  primes[1] = 0;

  for (i = 2; i <= Math.sqrt(n); i++) {
    if (primes[i] === 1) {
      for (let j = i; i * j <= n; j++) {
        primes[i * j] = 0;
      }
      // for (let j = i * 2; j <= n; j+=i) {
      //     primes[j] = 0;
      // }
    }
  }

  function getAllIndexes(arr, val) {
    var indexes = [],
      i;
    for (i = 0; i < arr.length; i++) if (arr[i] === val) indexes.push(i);
    return indexes;
  }

  const result = getAllIndexes(primes, 1);

  return result;
};

console.log(primeNumbers(2000000));
